app/models/nic/base.rb

```diff
@@ -210,16 +235,31 @@ def sync_name
       synchronizer.sync_name if synchronizer.sync_required?
     end
 
-    def validate_host_location
+    def validate_host_subnet_location
       return true if self.host.location.nil? && self.subnet.locations.empty?
       errors.add(:subnet, _("is not defined for host's location.")) unless include_or_empty?(self.subnet.locations, self.host.location)
     end
 
-    def validate_host_organization
+    def validate_host_subnet_organization
       return true if self.host.organization.nil? && self.subnet.organizations.empty?
       errors.add(:subnet, _("is not defined for host's organization.")) unless include_or_empty?(self.subnet.organizations, self.host.organization)
     end
 
+    def validate_host_subnet6_location
+      return true if self.host.location.nil? && self.subnet6.locations.empty?
+      errors.add(:subnet6, _("is not defined for host's location.")) unless include_or_empty?(self.subnet6.locations, self.host.location)
+    end
+
+    def validate_host_subnet6_organization
+      return true if self.host.organization.nil? && self.subnet6.organizations.empty?
+      errors.add(:subnet6, _("is not defined for host's organization.")) unless include_or_empty?(self.subnet6.organizations, self.host.organization)
+    end
+
+    def validate_subnets_types
+      errors.add(:subnet, _("must be of type Subnet::Ipv4.")) if self.subnet.present? && self.subnet.type != 'Subnet::Ipv4'
```

Comment:
I do think so. This validates, that the subnet for a nic is a ipv4 subnet and the subnet6 is a ipv6 subnet. The validation in your link checks that a subnet's type is either ipv4 or ipv6. So I think, both are required.